packages/user/AuthDyn/service/src/lib.rs

```diff
@@ -0,0 +1,245 @@
+#[psibase::service_tables]
+pub mod tables {
+    use async_graphql::SimpleObject;
+    use psibase::services::auth_dyn::int_wrapper;
+    use psibase::{
+        check, check_some, services::auth_dyn::interfaces::DynamicAuthPolicy, AccountNumber,
+        Fracpack, Table, ToSchema,
+    };
+    use serde::{Deserialize, Serialize};
+
+    #[table(name = "ManagementTable", index = 0)]
+    #[derive(Fracpack, ToSchema, SimpleObject, Serialize, Deserialize, Debug)]
+    pub struct Management {
+        #[primary_key]
+        pub account: AccountNumber,
+        pub manager: AccountNumber,
+    }
+
+    impl Management {
+        #[secondary_key(1)]
+        fn by_manager(&self) -> (AccountNumber, AccountNumber) {
+            (self.manager, self.account)
+        }
+
+        fn new(account: AccountNumber, manager: AccountNumber) -> Self {
+            Self { account, manager }
+        }
+
+        pub fn get(account: AccountNumber) -> Option<Self> {
+            ManagementTable::read().get_index_pk().get(&account)
+        }
+
+        pub fn get_assert(account: AccountNumber) -> Self {
+            check_some(Self::get(account), "failed to find management")
+        }
+
+        pub fn has_policy(&self) -> bool {
+            int_wrapper::call_to(self.manager).has_policy(self.account)
+        }
+
+        pub fn dynamic_policy(&self) -> Option<DynamicAuthPolicy> {
+            int_wrapper::call_to(self.manager).get_policy(self.account)
+        }
+
+        pub fn set(account: AccountNumber, policy: AccountNumber) -> Self {
+            let instance = Self::new(account, policy);
+            instance.save();
+            instance
+        }
+
+        pub fn assert_has_policy(&self) {
+            check(
+                self.has_policy(),
+                "management service does not support user",
+            )
+        }
+
+        fn save(&self) {
+            ManagementTable::read_write().put(self).unwrap();
+        }
+    }
+}
+
+#[psibase::service(name = "auth-dyn", tables = "tables", recursive = true)]
+pub mod service {
+    use crate::tables::Management;
+    use psibase::services::accounts::Wrapper as Accounts;
+    use psibase::services::auth_dyn::interfaces::{DynamicAuthPolicy, WeightedAuthorizer};
+    use psibase::services::transact::ServiceMethod;
+    use psibase::*;
+
+    #[action]
+    #[allow(non_snake_case)]
+    fn newAccount(account: AccountNumber) {
+        if let Some(management) = Management::get(account) {
+            check(
+                management.manager == get_sender(),
+                "new manager conflicts with pre-existing manager",
+            );
```

Comment:
At first it seems like this check isn't really necessary since it's a no-op. But I guess I could see an argument for keeping it in case there's a race between two different managers to create a particular account, you would want it to fail for the second manager... Is that your reasoning? Consider explaining that in a comment if so